api/tacticalrmm/clients/views.py

```diff
@@ -93,6 +93,10 @@ def post(self, request):
             core = get_core_settings()
             core.default_time_zone = request.data["timezone"]
             core.save(update_fields=["default_time_zone"])
+        # add in mesh company name
+        if "mesh_company_name" in request.data:
```

Comment:
1. no need to create a separate `if` block because we already have one to check if this is being called from `initialsetup`
2. `request.data["mesh_company_name"]` wont' work because on the frontend you are setting the variable to `companyname` not `mesh_company_name`
3. `update_fields_list` variable doesn't exist anywhere so you're trying to append to a non existing variable
4. no call to save in your new if block so even if it was right, would not actually get saved to db

here is how we can make it work using the existing `if` block:

```python
if "initialsetup" in request.data.keys():
    core = get_core_settings()
    core.default_time_zone = request.data["timezone"]
    core.mesh_company_name = request.data["companyname"]
    core.save(update_fields=["default_time_zone", "mesh_company_name"])
```